zcash_client_sqlite/src/wallet.rs

```diff
@@ -986,6 +1170,72 @@ pub(crate) fn block_height_extrema(
     })
 }
 
+pub(crate) fn get_account<C: Borrow<rusqlite::Connection>, P: Parameters>(
+    db: &WalletDb<C, P>,
+    account_id: AccountId,
+) -> Result<Option<Account>, SqliteClientError> {
+    let mut sql = db.conn.borrow().prepare_cached(
+        r#"
+        SELECT account_type, ufvk, uivk, hd_seed_fingerprint, hd_account_index
+        FROM accounts
+        WHERE id = :account_id
+    "#,
+    )?;
+
+    let mut result = sql.query(params![account_id.0])?;
+    let row = result.next()?;
+    match row {
+        Some(row) => {
+            let account_type: AccountType =
+                row.get::<_, u32>("account_type")?.try_into().map_err(|_| {
+                    SqliteClientError::CorruptedData("Unrecognized account_type".to_string())
+                })?;
+            let ufvk_str: Option<String> = row.get("ufvk")?;
+            let ufvk = if let Some(ufvk_str) = ufvk_str {
+                Some(
+                    UnifiedFullViewingKey::decode(&db.params, &ufvk_str[..])
+                        .map_err(SqliteClientError::BadAccountData)?,
+                )
+            } else {
+                None
+            };
```

Comment:
Rust note: this is more clean and less brittle as:
```suggestion
            let ufvk = ufvk_str.map(|ufvk_str| {
                UnifiedFullViewingKey::decode(&db.params, &ufvk_str[..])
                    .map_err(SqliteClientError::BadAccountData)?,
            });
```